subsys/bluetooth/host/gatt.c

```diff
@@ -180,13 +183,136 @@ BT_GATT_SERVICE_DEFINE(_2_gap_svc,
 #endif
 );
 
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
-static void sc_ccc_cfg_changed(const struct bt_gatt_attr *attr,
-			       u16_t value)
+#if defined(CONFIG_BT_GATT_SERVICE_CHANGED)
+struct gatt_sc_cfg {
+	u8_t                    id;
+	bt_addr_le_t		peer;
+	u8_t			data[4] __aligned(4);
+};
+
+#define SC_CFG_MAX (CONFIG_BT_MAX_PAIRED + CONFIG_BT_MAX_CONN)
+static struct gatt_sc_cfg sc_cfg[SC_CFG_MAX] = {};
+
+static struct gatt_sc_cfg *find_sc_cfg(u8_t id, bt_addr_le_t *addr)
+{
+	BT_DBG("id: %u, addr: %s", id, bt_addr_le_str(addr));
+
+	for (size_t i = 0; i < ARRAY_SIZE(sc_cfg); i++) {
+		if (id == sc_cfg[i].id &&
+		    !bt_addr_le_cmp(&sc_cfg[i].peer, addr)) {
+			return &sc_cfg[i];
+		}
+	}
+
+	return NULL;
+}
+
+struct sc_data {
+	u16_t start;
+	u16_t end;
+};
+
+#if defined(CONFIG_BT_SETTINGS)
+static int sc_store(const struct gatt_sc_cfg *cfg)
+{
+	char key[BT_SETTINGS_KEY_MAX];
+	char *data;
+	struct sc_data *sc;
+	size_t len;
+	int err;
+
+	if (cfg->id) {
+		char id_str[4];
+
+		u8_to_dec(id_str, sizeof(id_str), cfg->id);
+		bt_settings_encode_key(key, sizeof(key), "sc",
+				       (bt_addr_le_t *)&cfg->peer, id_str);
```

Comment:
The typecast seems unnecessary since cfg->peer is already a bt_addr_le_t

Reply:
Fixed in latest push